### PR TITLE
mon/MgrStatMonitor: keep mgrstat version ahead of pgmon

### PR DESCRIPTION
**Pull request by liewegas**
Pre-luminous, statfs and getpoolstats come from pgmon, and clients will
see and track the paxos version for that.  After the switchover, those
move to mgrstatmon.  If a clients sends a version ahead of the current
mgrstat version, it will block until the paxos version catches up, which
may be years for an aged cluster.

Fix by keeping mgrstat version ahead of pgmon's every time it updates.
This only really matters right at the transition point (we don't update
before that, and after that pgmon doesn't update).

Fixes: http://tracker.ceph.com/issues/20219
Signed-off-by: Sage Weil <sage@redhat.com>

qa
- [x] upgrade/jewel-x
- [x] rados

### CODE REVIEW

File: src/mon/MgrStatMonitor.cc [hunk 1]
```diff
@@ -100,6 +100,11 @@ void MgrStatMonitor::create_pending()
 void MgrStatMonitor::encode_pending(MonitorDBStore::TransactionRef t)
 {
   ++version;
+  if (version < mon->pgmon()->get_last_committed()) {
```
Comment (tchaikov) [inline src/mon/MgrStatMonitor.cc]: maybe we can just fix this in `MgrStatMonitor::create_initial()` instead to this check in every `encode_pending()`.

**Comment by liewegas** [comment]
The problem is that mgrstat is created when the mons become luminous 
(upgrade starts), but the switchover happens later when the osdmap 
require_osd_release is updated (upgrade finishes).

**Comment by liewegas** [comment]
http://pulpito.ceph.com/sage-2017-06-11_18:26:58-rados-wip-sage-testing2-distro-basic-smithi/

**Comment by liewegas** [comment]
http://pulpito.ceph.com/sage-2017-06-12_13:49:59-upgrade:jewel-x-wip-sage-testing2---basic-smithi/

**Pull request merged**